pkg/database/database.go

```diff
@@ -121,46 +141,55 @@ func (db *Database) QueryLogURLs(ctx context.Context, kind, apiVersion, namespac
 	uuid, err := strQueryPerformer.performSingleRowQuery(ctx, sb)
 
 	if errors.Is(err, sql.ErrNoRows) {
-		return []string{}, ResourceNotFoundError
+		return "", "", ResourceNotFoundError
 	}
 
 	if err != nil {
-		return nil, err
+		return "", "", err
 	}
 
-	podUuids, err := db.getOwnedPodsUuids(ctx, []string{uuid}, []string{})
-	if len(podUuids) == 0 {
-		return []string{}, ResourceNotFoundError
-	}
+	pods, err := db.getOwnedPodsUuids(ctx, []string{uuid}, []uuidKindDate{})
 	if err != nil {
-		return nil, err
+		return "", "", err
 	}
+	if len(pods) == 0 {
+		return "", "", ResourceNotFoundError
+	}
+
+	// Get the most recent pod from owned by the provided resource
+	sort.Slice(pods, func(i, j int) bool {
+		return pods[i].Date > pods[j].Date
+	})
```

Comment:
the docs for [`sort.Slice`](https://pkg.go.dev/sort#Slice) says that [`slices.SortFunc`](https://pkg.go.dev/slices#SortFunc) is generally faster. It might be worth it instead

Reply:
I haven't noticed that suggestion! Thanks for the pointers, the change is already in place :)